src/supertux/gameconfig.cpp

```diff
@@ -205,13 +229,19 @@ Config::save()
 
   writer.start_list("control");
   {
-    writer.start_list("keymap");
-    keyboard_config.write(writer);
-    writer.end_list("keymap");
+    assert(keyboard_configs.size() == joystick_configs.size());
```

Comment:
Can a malformed config file make this assert fail? If so, please exit more gracefully/provide a warning and initialize new default keyboards? 
(I am currently not sure if asserts are enabled in release mode @tobbi)